src/components/SignIn.jsx

```diff
@@ -0,0 +1,56 @@
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { getAllUsers } from "../../utils";
+
+export function SignIn({ setUser }) {
+  const [allUsers, setAllUsers] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    getAllUsers().then(({ users }) => {
+      setAllUsers(users);
+      setLoading(false);
+    });
+  }, []);
+
+  function handleClick(user) {
+    setUser(user);
+    alert(`You have signed-in as ${user.username}`);
+    navigate(-1);
```

Comment:
I really like this little touch, which is it takes the user to the previous page, back to where they were, after they sign in - love it